src/ops/delay.c

```diff
@@ -42,3 +65,67 @@ static void op_DEL_CLR_get(const void *NOTUSED(data), scene_state_t *ss,
                            command_state_t *NOTUSED(cs)) {
     clear_delays(ss);
 }
+
+static void mod_DEL_X_func(scene_state_t *ss, exec_state_t *es,
+                           command_state_t *cs,
+                           const tele_command_t *post_command) {
+    int16_t i = 0;
+    int16_t num_delays = cs_pop(cs);
+    int16_t delay_time = cs_pop(cs);
+    int16_t delay_time_next;
+
+    if (delay_time < 1) delay_time = 1;  // minimum delay time = 1ms
+
+    delay_time_next = delay_time;  // set first delay time to delay time
+
+    // 0 is the magic number for an empty slot.
+    // Be careful not to set delay.time[i] to 0 before calling this function.
+    while (ss->delay.time[i] != 0 && i != DELAY_SIZE) i++;
+
+    while (i < DELAY_SIZE && num_delays > 0) {
+        // queue the delay
+        delay_common_add(ss, es, i, delay_time_next, post_command);
+
+        // increment delay time for next delay
+        // normalise incremented value to stop negative wrap from increment
+        delay_time_next += delay_time;
+        delay_time_next = normalise_value(1, 32767, 1, delay_time_next);
+
+        num_delays--;
+        i++;
```

Comment:
sorry, should've looked into the code closer the 1st time.. there is a problem with this spot here. for multiple delays it will find the 1st available slot but for any consecutive delays it will use the next delay without checking if it's free or not. you have to check that for each delay added. 

this: `while (ss->delay.time[i] != 0 && i != DELAY_SIZE) i++;` can be moved to `delay_common_add` and `i` parameter can be removed, so the function's responsibility will be finding the 1st available spot and adding a delay there. it could return a bool indicating if it was added, so if `false` is returned this loop can assume no more spots are available and exit early.

Reply:
wow another good catch.  elegant solution too.  I'll work on getting that together tonight.